core/src/main/java/org/axonframework/queryhandling/SimpleQueryBus.java

```diff
@@ -73,72 +66,41 @@
  * @author Milan Savic
  * @since 3.1
  */
-public class SimpleQueryBus implements QueryBus, QueryUpdateEmitter {
+public class SimpleQueryBus implements QueryBus {
 
     private static final Logger logger = LoggerFactory.getLogger(SimpleQueryBus.class);
 
-    private static final String QUERY_UPDATE_TASKS_RESOURCE_KEY = "/update-tasks";
-
     private final ConcurrentMap<String, CopyOnWriteArrayList<QuerySubscription>> subscriptions = new ConcurrentHashMap<>();
-    private final ConcurrentMap<SubscriptionQueryMessage<?, ?, ?>, FluxSinkWrapper<?>> updateHandlers = new ConcurrentHashMap<>();
     private final MessageMonitor<? super QueryMessage<?, ?>> messageMonitor;
-    private final MessageMonitor<? super SubscriptionQueryUpdateMessage<?>> updateMessageMonitor;
     private final QueryInvocationErrorHandler errorHandler;
     private final List<MessageHandlerInterceptor<? super QueryMessage<?, ?>>> handlerInterceptors = new CopyOnWriteArrayList<>();
     private final List<MessageDispatchInterceptor<? super QueryMessage<?, ?>>> dispatchInterceptors = new CopyOnWriteArrayList<>();
 
-    /**
-     * Initialize the query bus without monitoring on messages and a {@link LoggingQueryInvocationErrorHandler}.
-     */
-    public SimpleQueryBus() {
-        this(NoOpMessageMonitor.INSTANCE, NoTransactionManager.instance(),
-             new LoggingQueryInvocationErrorHandler(logger));
-    }
-
-    /**
-     * Initialize the query bus using given {@code transactionManager} to manage transactions around query execution
-     * with. No monitoring is applied to messages and a {@link LoggingQueryInvocationErrorHandler} is used
-     * to log errors on handlers during a scatter-gather query.
-     *
-     * @param transactionManager The transaction manager to manage transactions around query execution with
-     */
-    public SimpleQueryBus(TransactionManager transactionManager) {
-        this(NoOpMessageMonitor.INSTANCE, transactionManager, new LoggingQueryInvocationErrorHandler(logger));
-    }
+    private final QueryUpdateEmitter queryUpdateEmitter;
 
     /**
-     * Initialize the query bus with the given {@code messageMonitor} and given {@code errorHandler}.
-     *
-     * @param messageMonitor     The message monitor notified for incoming messages and their result
-     * @param transactionManager The transaction manager to manage transactions around query execution with
-     * @param errorHandler       The error handler to invoke when query handler report an error
+     * Initialize the query bus without monitoring on messages and a {@link LoggingQueryInvocationErrorHandler}.
```

Comment:
I think this javadoc is a bit out of sync with what this constructor actually does.
Would it make sense to keep the javadoc for all the protected constructors which take in a builder somewhat identical? So also explaining which fields are validated and such?

Reply:
Ah, so I see nothing is validated on the `validate()` function...fair, then that point goes away.
Nonetheless I still think some important things have been omitted from this javadoc.